tests/core/test_dihypergraph.py

```diff
@@ -280,7 +280,7 @@ def test_add_edges_from_format4():
     H1 = xgi.DiHypergraph([({1, 2}, {3}), ({2, 3, 4}, {1})])
     with pytest.warns(Warning):
         H1.add_edges_from([(({0, 1}, {2}), 0, {"color": "red"})])
-    assert H1._edge_out == {0: {1, 2}, 1: {2, 3, 4}}
+    assert {i: e["out"] for i, e in H1._edge.items()} == {0: {3}, 1: {1}}
```

Comment:
this looks more complicated to access the edges out than before (`H1._edge_out`). Could we do something like `H._edge["out"]` ?

Reply:
So this won't change the API, but I get what you mean. Because I chose an edge/node ID-first approach, when trying to get just the inputs or outputs, it will require some kind of list/dict comprehension.